Sources/Benchmark/main.swift

```diff
@@ -0,0 +1,330 @@
+}
+
+try time("NSRegex #6") {
+  try run_simple_alternation_nsregex(N)
+}
+
+print()
+print(#"Simple quote match: 'a\Q \Q \\.\Eb' in '123a \Q \\.bxyz'"#)
+
+@inline(never)
+func run_simple_quotes_native(_ N: Int) throws {
+  for _ in 0 ..< N {
+    let match = try nativeMatch(#"a\Q \Q \\.\Eb"#, input: #"123a \Q \\.bxyz"#)
+    
+    blackHole(match)
+  }
+}
+
+@inline(never)
+func run_simple_quotes_nsregex(_ N: Int) throws {
+  for _ in 0 ..< N {
+    let match = try nsregexMatch(#"a\Q \Q \\.\Eb"#, input: #"123a \Q \\.bxyz"#)
+    
+    blackHole(match)
+  }
+}
+
+try time("Native #7") {
+  try run_simple_quotes_native(N)
+}
+
+try time("NSRegex #8") {
+  try run_simple_quotes_nsregex(N)
+}
+
+print()
+print(#"Simple character name match: '\N{ASTERISK}' in '123***xyz'"#)
+
+@inline(never)
+func run_simple_character_name_native(_ N: Int) throws {
+  for _ in 0 ..< N {
+    let match = try nativeMatch(#"\N{ASTERISK}"#, input: "123***xyz")
+    
+    blackHole(match)
+  }
+}
+
+@inline(never)
+func run_simple_character_name_nsregex(_ N: Int) throws {
+  for _ in 0 ..< N {
+    let match = try nsregexMatch(#"\N{ASTERISK}"#, input: "123***xyz")
+    
+    blackHole(match)
+  }
+}
+
+try time("Native #9") {
+  try run_simple_character_name_native(N)
+}
+
+try time("NSRegex #10") {
+  try run_simple_character_name_nsregex(N)
+}
+
+print()
+print(#"Simple script match: '\P{Script=Latn}' in 'abcαβγxyz'"#)
+
+@inline(never)
+func run_simple_script_native(_ N: Int) throws {
+  for _ in 0 ..< N {
+    let match = try nativeMatch(#"\P{Script=Latn}"#, input: "abcαβγxyz")
+    
+    blackHole(match)
+  }
+}
+
+@inline(never)
+func run_simple_script_nsregex(_ N: Int) throws {
+  for _ in 0 ..< N {
+    let match = try nsregexMatch(#"\P{Script=Latn}"#, input: "abcαβγxyz")
+    
+    blackHole(match)
+  }
+}
+
+try time("Native #11") {
+  try run_simple_script_native(N)
+}
+
+try time("NSRegex #12") {
+  try run_simple_script_nsregex(N)
+}
+
+print()
+print(#"Simple group match: '(?:a|.b)c' in '123abcxyz'"#)
+
+@inline(never)
+func run_simple_group_native(_ N: Int) throws {
+  for _ in 0 ..< N {
+    let match = try nativeMatch(#"(?:a|.b)c"#, input: "123abcxyz")
+    
+    blackHole(match)
+  }
+}
+
+@inline(never)
+func run_simple_group_nsregex(_ N: Int) throws {
+  for _ in 0 ..< N {
+    let match = try nsregexMatch(#"(?:a|.b)c"#, input: "123abcxyz")
+    
+    blackHole(match)
+  }
+}
+
+try time("Native #13") {
+  try run_simple_group_native(N)
+}
+
+try time("NSRegex #14") {
+  try run_simple_group_nsregex(N)
+}
+
+print()
+print(#"Simple comment match: 'a(?#. comment)b' in '123abcxyz'"#)
+
+@inline(never)
+func run_simple_comment_native(_ N: Int) throws {
+  for _ in 0 ..< N {
+    let match = try nativeMatch(#"a(?#. comment)b"#, input: "123abcxyz")
+    
+    blackHole(match)
+  }
+}
+
+@inline(never)
+func run_simple_comment_nsregex(_ N: Int) throws {
+  for _ in 0 ..< N {
+    let match = try nsregexMatch(#"a(?#. comment)b"#, input: "123abcxyz")
```

Comment:
Do we want a function that scales up the input length (so we have longer inputs and fewer compilations)? Similarly, do we want to have all-ranges style benchmarks?